### DISCUSSION TITLE
getActiveAccount() and setActiveAccount() APIs

### DISCUSSION THREAD

**Discussion by arsdragonfly** [AzureAD/microsoft-authentication-library-for-python#615] [Ideas]
Is it possible to add something like [this](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/accounts.md#active-account-apis) to MSAL for Python? I'd also like to see a way to obtain an enriched response with account information computed from the raw JSON response, like the way it's done in MSAL for JS.

**Comment by bgavrilMS** [comment]
Higher level APIs are not on the roadmap for public client applications now, only for confidential client applications. 

The functionality is very simple though - just assume there can only be 1 account in the cache. Operation is always in the order: 

- get account from cache. If more than 1 is found, bail (or wipe the cache and start again)
- if 0 accounts found, acquire token interactive or acquire token for Windows user
- if 1 account found, try to acquire token silently. If that fails (e.g. MFA is required), acquire token interactive with login hint.

Not sure I get the "raw json response" idea.
